packages/nimble-components/src/anchor-button/tests/anchor-button.spec.ts

```diff
@@ -0,0 +1,298 @@
+import { DOM, html } from '@microsoft/fast-element';
+import { AnchorButton } from '..';
+import { fixture, Fixture } from '../../utilities/tests/fixture';
+
+async function setup(): Promise<Fixture<AnchorButton>> {
+    return fixture<AnchorButton>(
+        html`<nimble-anchor-button></nimble-anchor-button>`
+    );
+}
+
+describe('AnchorButton', () => {
+    let element: AnchorButton;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('can construct an element instance', () => {
+        expect(document.createElement('nimble-anchor-button')).toBeInstanceOf(
+            AnchorButton
+        );
+    });
+
+    it('should set the "control" class on the internal control', async () => {
+        await connect();
+        expect(element.control.classList.contains('control')).toBe(true);
+    });
+
+    it('should set the `part` attribute to "control" on the internal control', async () => {
+        await connect();
+        expect(element.control.part.contains('control')).toBe(true);
+    });
+
+    it('should clear `href` on the internal control when disabled', async () => {
+        await connect();
+        element.control.setAttribute('href', 'http://www.ni.com');
+
+        element.disabled = true;
+        await DOM.nextUpdate();
+
+        expect(element.control.getAttribute('href')).toBeNull();
+    });
+
+    it('should reflect `download` value to the internal control', async () => {
```

Comment:
As I asked in anchor tests, should we adopt a theory pattern for most of these tests?

Reply:
Done